worlds/stardew_valley/test/rules/TestSkills.py

```diff
@@ -25,16 +25,56 @@ def test_all_skill_levels_require_previous_level(self):
         for skill in all_skills:
             self.collect_everything()
             self.remove_by_name(f"{skill} Level")
+
             for level in range(1, 11):
                 location_name = f"Level {level} {skill}"
+                location = self.multiworld.get_location(location_name, self.player)
+
                 with self.subTest(location_name):
-                    can_reach = self.can_reach_location(location_name)
                     if level > 1:
-                        self.assertFalse(can_reach)
+                        self.assert_reach_location_false(location, self.multiworld.state)
                         self.collect(f"{skill} Level")
-                        can_reach = self.can_reach_location(location_name)
-                    self.assertTrue(can_reach)
-            self.multiworld.state = self.original_state.copy()
 
+                    self.assert_reach_location_true(location, self.multiworld.state)
+
+            self.reset_collection()
+
+
+class TestMasteryRequireSkillBeingMaxed(SVTestBase):
+    #  Using minsanity so collecting everything is faster
+    options = get_minsanity_options()
+    options.update({
+        SkillProgression.internal_name: SkillProgression.option_progressive_with_masteries,
+        Mods.internal_name: frozenset(),
```

Comment:
Is this needed?

Reply:
No, but yes because I'll remove the minsanity options to use the world cache, so it's necessary to not get a world with mods from the cache.